chore: Update docs to add universal RPC info (#1156)
Unity-Technologies/com.unity.multiplayer.docs: docs/advanced-topics/message-system/rpc-compatibility.md

## Cross-Compatibility

Learn more about standard RPC API's cross-compatibility only, not the framework as a whole. A method decorated with an RPC attribute will be statically registered with its assembly-scoped method signature hash.

A typical assembly-scoped method signature sample:

```
Game.dll / System.Void Shooter::PingRpc(System.Int32)
```

where:

- `Game.dll` is the Assembly
- `/` is a Separator
- `System.Void Shooter::PingRpc(System.Int32)` is the Method signature:

    - `System.Void` is the Return type
    - `Shooter` is the Enclosing type
    - `::` is the Scope resolution operator
    - `PingRpc` is the Method name
    - `(System.Int32)` is the Parameter type (no param names)

An RPC signature will be turned into a 32-bit integer using [xxHash](https://cyan4973.github.io/xxHash/) (XXH32) non-cryptographic hash algorithm.

As such, if the assembly, enclosing type, method name and/or any method param type changes then the RPC signature hash will change. However, if only the names of the parameters for an existing RPC change it will not change the method signature.

> Since changing the RPC signature leads into a different invocation code path that has different serialization code, previous versions of that RPC method (i.e. older builds) will not be remotely invoked by the RPC method with the new signature.

Author: Not at all. Basically the old RPC no longer exists, there's no way for the other side to find it.

> Since changing the RPC signature leads into a different invocation code path that has different serialization code, previous versions of that RPC method (i.e. older builds) will not be remotely invoked by the RPC method with the new signature.

Reviewer: Good to know, thanks!

| Compatibility | <i class="fp-check"></i> | Description |
| -- | :--: | -- |
| Cross-Build Compatibility | <i class="fp-check"></i> | As long as the RPC method signature is kept the same, it will be compatible between different builds. |
| Cross-Version Compatibility | <i class="fp-check"></i> | As long as the RPC method signature is kept the same, it will be compatible between different versions. |

> | Cross-Project Compatibility | <i class="fp-x"></i> | Since the project name or project-specific token isn't part of RPC signature, it is possible to have the exact same RPC method signature defined in different projects. Cross project RPC methods will not be be compatible with each other. |

Reviewer: ```suggestion
| Cross-Project Compatibility | <i class="fp-x"></i> | The exact same RPC method signature can be defined in different projects. This is because the project name or project-specific token isn't part of RPC signature. Cross-project RPC methods aren't compatible with each other. |
```
